tar/bsdtar.c

```diff
@@ -416,7 +417,9 @@ main(int argc, char **argv)
 			optq_push(bsdtar, "exclude", bsdtar->optarg);
 			break;
 		case 'f': /* multitar */
-			bsdtar->tapenames[bsdtar->ntapes++] = bsdtar->optarg;
+			tapename_cmdline = (char *)bsdtar->optarg;
```

Comment:
How does this even compile?  `bsdtar->optarg` is a `const char *`... you can't just cast away `const` like that!